x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/source_router.tsx

```diff
@@ -61,84 +48,43 @@ export const SourceRouter: React.FC = () => {
     return resetSourceState;
   }, []);
 
-  if (dataLoading) return <Loading />;
+  if (dataLoading) {
+    return isOrganization ? (
+      <WorkplaceSearchPageTemplate isLoading />
+    ) : (
+      <PersonalDashboardLayout isLoading />
+    );
+  }
```

Comment:
NOTE: This early page isLoading behavior is required to prevent a flash of a completely empty page (instead of preserving the layout/side nav while loading). We cannot let the page fall through to the route because some routes are conditionally rendered based on `isCustomSource` below.

FWIW: App Search will have a similar page isLoading early return with its Engine sub nav, so it's not a completely unusual pattern - it mostly happens when a certain data type needs to be loaded w/ preliminary data before its routing or side nav.

Reply:
Love this pattern!